modules/rubiconBidAdapter.js

```diff
@@ -949,6 +868,81 @@ function addVideoParameters(data, bidRequest) {
   data.imp[0].video.h = size[1]
 }
 
+function applyFPD(bidRequest, mediaType, data) {
+  const bidFpd = {
+    user: {...bidRequest.params.visitor},
+    context: {...bidRequest.params.inventory}
+  };
+
+  if (bidRequest.params.keywords) bidFpd.context.keywords = (utils.isArray(bidRequest.params.keywords)) ? bidRequest.params.keywords.join(',') : bidRequest.params.keywords;
+
+  let fpd = utils.mergeDeep({}, config.getConfig('fpd') || {}, bidRequest.fpd || {}, bidFpd);
```

Comment:
Unfortunately, I also think we need to merge adunit/imp-level FPD into imp[].ext.data rather than {site,user}.ext.data. 

And we need to take special care with:
- AdUnit.fpd.context.pbAdSlot --> openrtb imp[].ext.data.pbadslot and fastlane tg_i.pbadslot
- ortb2Imp.ext.data.pbAdSlot --> openrtb imp[].ext.data.pbadslot and fastlane tg_i.pbadslot
- AdUnit.fpd.context.adServer.adSlot --> openrtb imp[].ext.data.adserver.adslot and fastlane tg_i.dfp_ad_unit_code
- ortb2Imp.ext.data.adServer.adSlot --> openrtb imp[].ext.data.adserver.adslot and fastlane tg_i.dfp_ad_unit_code

If the ortb2 changes aren't in scope for this PR, that's cool. I know there's a different PR for that beast.

Reply:
Made a test page and this all actually does work because it's a single-adunit request for a single bidder. Going to resolve these issues.